tests/pkg_uzlib/main.c

```diff
@@ -0,0 +1,113 @@
+ * number of possible values (1 << hash_bits, with hash_bits set to 6 in
+ * this application) */
+#define HASH_TABLE_SIZE     (8 * 64)
+#define CHUNK_SIZE          (1)
+static uzlib_hash_entry_t hash_table[HASH_TABLE_SIZE];
+
+static const char *message =    "Lorem ipsum dolor sit amet, consectetur "
+                                "adipiscing elit, sed do eiusmod tempor "
+                                "incididunt ut labore et dolore magna aliqua. "
+                                "Nisl vel pretium lectus quam id leo. Volutpat "
+                                "sed cras ornare arcu dui vivamus arcu felis "
+                                "bibendum. Velit scelerisque in dictum non "
+                                "consectetur a erat nam. Pretium viverra "
+                                "suspendisse potenti nullam ac tortor vitae "
+                                "purus faucibus. Tristique et egestas quis "
+                                "ipsum suspendisse. At imperdiet dui accumsan "
+                                "sit amet nulla facilisi. Pulvinar neque "
+                                "laoreet suspendisse interdum consectetur "
+                                "libero. Vulputate sapien nec sagittis aliquam "
+                                "malesuada bibendum arcu vitae. Sed adipiscing "
+                                "diam donec adipiscing tristique risus nec. "
+                                "Venenatis tellus in metus vulputate eu "
+                                "scelerisque. Id faucibus nisl tincidunt eget "
+                                "nullam non nisi est. Integer feugiat "
+                                "scelerisque varus morbi enim. Est sit amet "
+                                "facilisis magna etiam. Venenatis cras sed "
+                                "felis eget velit aliquet sagittis.";
+static unsigned char decompressed_output[1024];
+
+int main(void)
+{
+    /* Compress the message */
+    struct uzlib_comp comp = {0};
+    comp.dict_size = 32768;
+    comp.hash_bits = 6;
+    comp.hash_table = hash_table;
+
+    unsigned message_len = strlen(message);
+
+    zlib_start_block(&comp.out);
+    uzlib_compress(&comp, (uint8_t *)message, message_len);
+    zlib_finish_block(&comp.out);
+
+    unsigned compression_ratio = (comp.out.outlen * 10000) / message_len;
+    printf("Compressing message (len: %u): \"%s\"\n", message_len, message);
+    printf("Message compressed to %u raw bytes (ratio: %u.%u)\n",
+           comp.out.outlen,
+           compression_ratio / 10000, (compression_ratio / 100) % 100);
+
+    /* Decompress */
+    uzlib_init();
+    unsigned int dlen;
+
+    /* Get decompressed length */
+    dlen =            comp.out.outbuf[comp.out.outlen - 1];
+    dlen = 256 * dlen + comp.out.outbuf[comp.out.outlen - 2];
+    dlen = 256 * dlen + comp.out.outbuf[comp.out.outlen - 3];
+    dlen = 256 * dlen + comp.out.outbuf[comp.out.outlen - 4];
```

Comment:
```suggestion
    dlen = unaligned_get_u32(&comp.out.outbuf[comp.out.outlen - 4]);
```

If that's not in host byte order, there is `byteorder_bebuftohl()`

Reply:
Thanks, that's more concise.